stdlib/printexc.ml

```diff
@@ -69,6 +69,11 @@ let to_string x =
               constructor ^ (fields x) in
   conv !printers
 
+let to_string_opt x =
+  match !printers with
+      [] -> None
+    | _ -> Some (to_string x)
+
```

Comment:
This is not the right semantics: `!printers <> []` does not imply that there is a printer for `x` in `!printers`. Rather, I suggest:

1) Rename `to_string` into `to_string_opt` and change the code so that it returns an option with `None` in the case where there is no registered printer (right now it uses `Obj.magic` to produce a representation in this case).

2)  Re-implement `to_string` in terms of `to_string_opt` (the previous `Obj.magic` code would be moved here to handle the `None` case).